frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx

```diff
@@ -0,0 +1,196 @@
+import { useEffect, useMemo } from 'react'
+import { Controller, RegisterOptions } from 'react-hook-form'
+import { FormControl, SimpleGrid } from '@chakra-ui/react'
+import { extend, isEmpty, pick } from 'lodash'
+
+import { NumberFieldBase, NumberSelectedValidation } from '~shared/types/field'
+
+import { createBaseValidationRules } from '~utils/fieldValidation'
+import { SingleSelect } from '~components/Dropdown'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import NumberInput from '~components/NumberInput'
+import Textarea from '~components/Textarea'
+import Toggle from '~components/Toggle'
+
+import { DrawerContentContainer } from '../common/DrawerContentContainer'
+import { FormFieldDrawerActions } from '../common/FormFieldDrawerActions'
+import { EditFieldProps } from '../common/types'
+import { useEditFieldForm } from '../common/useEditFieldForm'
+
+type EditNumberProps = EditFieldProps<NumberFieldBase>
+
+const EDIT_NUMBER_FIELD_KEYS = ['title', 'description', 'required'] as const
+
+type EditNumberInputs = Pick<
+  NumberFieldBase,
+  typeof EDIT_NUMBER_FIELD_KEYS[number]
+> & {
+  ValidationOptions: {
+    selectedValidation: NumberSelectedValidation | ''
+    customVal: number | ''
+  }
+}
+
+const transformNumberFieldToEditForm = (
+  field: NumberFieldBase,
+): EditNumberInputs => {
+  const nextValidationOptions = {
+    selectedValidation:
+      field.ValidationOptions.selectedValidation || ('' as const),
+    customVal:
+      (!!field.ValidationOptions.selectedValidation &&
+        field.ValidationOptions.customVal) ||
+      ('' as const),
+  }
+  return {
+    ...pick(field, EDIT_NUMBER_FIELD_KEYS),
+    ValidationOptions: nextValidationOptions,
+  }
+}
+
+const transformNumberEditFormToField = (
+  form: EditNumberInputs,
+  originalField: NumberFieldBase,
+): NumberFieldBase => {
+  const nextValidationOptions =
+    form.ValidationOptions.selectedValidation === ''
+      ? {
+          selectedValidation: null,
+          customVal: null,
+        }
+      : form.ValidationOptions
+  return extend({}, originalField, form, {
```

Comment:
is `extend()` overreaching here? What inherited properties do we need from `originalField` and `form`?

Reply:
it's fine I think, `form` is actually the updated inputs which we want to override with (I'll update the naming actually for clarity). The return type of `NumberFieldBase` allows us to be certain that the shape has not changed